openproblems/tasks/spatial_decomposition/datasets/_destvi_utils.py

```diff
@@ -20,14 +23,44 @@
 torch.manual_seed(0)
 
 
-def generate_synthetic_dataset(
+def categorical(p, n_samples):
+    size = list(p.shape[:-1])
+    size.insert(0, n_samples)
+    return (p.cumsum(-1) >= np.random.uniform(size=size)[..., None]).argmax(-1).T
+
+
+@jit(nopython=True)
+def get_mean_normal(cell_types, gamma, mean_, components_):
```

Comment:
Add `# pragma: no cover` to skip coverage on JIT compiled code. See https://github.com/openproblems-bio/openproblems/blob/main/openproblems/tasks/dimensionality_reduction/metrics/nn_ranking.py for an example